docs/docs/tutorials/rag/index.ipynb

```diff
@@ -67,67 +67,160 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": 2,
    "metadata": {},
    "outputs": [],
    "source": [
     "import dspy\n",
+    "import os\n",
+    "\n",
+    "os.environ[\"OPENAI_API_KEY\"] = \"{your_openai_api_key}\"\n",
     "\n",
-    "lm = dspy.LM('openai/gpt-4o-mini')\n",
+    "lm = dspy.LM(\"openai/gpt-4o-mini\")\n",
     "dspy.configure(lm=lm)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Exploring some basic DSPy Modules.\n",
+    "## Basic Retrieval-Augmented Generation (RAG)\n",
+    "\n",
+    "First, let's download the corpus data that we will use for RAG search. An older version of this tutorial used the full (650,000 document) corpus. To make this very fast and cheap to run, we've downsampled the corpus to just 28,000 documents."
```

Comment:
nit: we might not need to mention "An older version of this tutorial used the full (650,000 document) corpus." since I think many users will see the page for the first time